app/views/comments/_hide_comments_modal.html.erb

```diff
@@ -0,0 +1,19 @@
+<div id="hide-comments-modal" class="hidden">
+  <%= form_with url: "/comments/hide", class: "hide-comments-modal__form", id: "hide-comments-modal__form", method: :patch, remote: true, html: { "data-type": "json" } do |f| %>
+    <div class="hide-comments-modal__content">
+      <p class="pb-2">
+        Are you sure you want to hide this comment? It will become hidden in your post, but will still be visible via the comment's permalink.
+      </p>
+      <label class="crayons-field crayons-field--checkbox my-2">
+        <%= f.check_box :hide_children, class: "hide_children crayons-checkbox" %>
+        <p class="crayons-field__label">Hide child comments as well</p>
+      </label>
+      <p class="pb-4 pt-2">
+        <button type="submit" class="crayons-btn" aria-label="<%= t("views.comments.hide.confirm") %>">
```

Comment:
(accessibility / not blocking) - for what it's worth, I don't think the aria-label is necessary here. There's plenty context in the modal, and the text "Confirm" would be enough (no harm either way though)

Reply:
Actually, I added `aria-label` to use it in a cypress test :thinking:  I suppose it doesn't make sense to keep the `aria-label` but simplify its text, or does it?

Reply:
I think you should be able to grab a reference to the button in Cypress without the aria-label - e.g. `cy.findByRole('button', { name: 'Confirm' })` 🤔  If that's not specific enough you could also do `cy.findByTestId('modal-container').findByRole('button', { name: 'Confirm' })` so it only matches the confirm button presented in the modal.

I generally try not to add ARIA for testing purposes, just because it's kind of 'invisible' but does affect screen reader users - it's quite easy for us to miss e.g. the aria-label and the visible label diverging in future because we're not always testing with screen readers

Reply:
Thanks for the explanation, that makes sense :sun_with_face: